server/controller/apis/topicController.js

```diff
@@ -278,33 +247,83 @@ exports.saveTopic = async ( req, res, redirect ) => {
 
         if(topic.topicType == 3) {
             
-            topic.topicContentHtml = req.body.embedded_submission_text_topic;
+            topic.topicHtml = req.body.embedded_submission_text_topic;
         }
         else {
             // check to see if the incomping message format is from the UI or the API
-            if( req.body.topicContentHtml ) {
-                topic.topicContentHtml = req.body.topicContentHtml;
+            if( req.body.topicHtml ) {
+                topic.topicHtml = req.body.topicHtml;
             }
             else {
-                topic.topicContentHtml = req.body.topicEditor;
+                topic.topicHtml = req.body.topicEditor;
             }
         }
-        topic.topicLink = req.body.topicLink;
         
         // check to see if the incoming message format is from the UI form or the API
-        if( req.body.topicActive ) {
-            topic.active = ( req.body.topicActive == "on" ) ? true : false;
-        }
-        else if ( req.body.active ) {
+        topic.active = false; // Defaulted to false if not specified.
+        if( req.body.active ) {
             topic.active = req.body.active;
         }
```

Comment:
Similar to my concern from new lines 254-258, the previous code checked for a field topicActive (see line 87 of client/agora/views/dashboard/partials/topic/edit-topic.ejs). We should leave this check in place to allow the front-end code to remain unchanged.